src/Tasks/StateFileBase.cs

```diff
@@ -77,32 +78,35 @@ internal static StateFileBase DeserializeCache(string stateFile, TaskLoggingHelp
                     using (FileStream s = new FileStream(stateFile, FileMode.Open))
                     {
                         var formatter = new BinaryFormatter();
```

Comment:
The commit history shows you reverted back to a binaryformatter over json serialization. What was the reason for this again?

Reply:
Had trouble getting it to work. Made it work in #5704.